src/pmdas/gfs2/glockfd.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * GFS2 glockfd file statistics.
+ *
+ * Copyright (c) 2013-2025 Red Hat.
+ * 
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef GLOCKFD_H
+#define GLOCKFD_H
+
+#define MAX_NUMBER_OF_HOLDERS 10000
+
+enum {
+    GLOCKFD_TOTAL = 0,
+    NUM_GLOCKFD_STATS
+};
+
+enum {
+    GLOCKFD_PROCESS = 0,
+    GLOCKFD_FILE_DESCRIPTOR,
+    NUM_PER_HOLDER_STATS
+};
+
+struct glockfd {
+    __uint64_t total;
+    __uint64_t process[MAX_NUMBER_OF_HOLDERS];
+    __uint64_t file_descriptor[MAX_NUMBER_OF_HOLDERS];
```

Comment:
32 bits for the second two fields please.  And, hmm, is this a long-lived structure?  Seems slightly wasteful of memory if so (alternatively, may be a target for buffer overrun) - can we dynamically size these arrays.